.github/actions/cargo-semver-checks/action.yml

```diff
@@ -0,0 +1,24 @@
+name: "Run cargo semver-checks"
+description: "Install and run the cargo semver-checks tool"
+inputs:
+  args:
+    required: true
+    description: "The arguments to pass to `cargo semver-checks`"
+runs:
+  using: "composite"
+  steps:
+    - run: wget -q -O- https://github.com/obi1kenobi/cargo-semver-checks/releases/download/v0.18.3/cargo-semver-checks-x86_64-unknown-linux-gnu.tar.gz | tar -xz -C ~/.cargo/bin
+      shell: bash
+
+    - uses: Swatinem/rust-cache@359a70e43a0bb8a13953b04a90f76428b4959bb6 # v2.2.0
+      with:
+        shared-key: semver-checks-cache
+        save-if: ${{ github.ref_name == 'master' }}
+        workspaces: |
+          .
+          target/semver-checks
```

Comment:
As far as I can tell, this doesn't help at all right now — it ends up rebuilding everything for the "current" rustdoc, and you can confirm this by setting `CARGO_TERM_VERBOSE=true` and observing that compilations are happening instead of being shown as skipped due to `Fresh` build data.

I know that cargo uses mtimes to decide whether to rebuild or not. I tried setting a synthetic mtime in the future on the restored build artifacts, but even that didn't seem to help: https://github.com/obi1kenobi/cargo-semver-checks/pull/399

I don't think I have the cycles to continue digging into the mtimes issue right now, but I thought I'd give you a heads-up in case you're able to dig further.

Reply:
Hmm, thanks for that! Yeah I did wonder why this isn't working as expected. I may resort to only caching the registry JSONs files for now.

It is a bit annoying because this job by itself takes almost an hour for us and would run on every PR merged into master with most of the compute being wasted if I can't cache the debug symbols for all the crates.

Reply:
Yeah, I feel that :( I opened https://github.com/obi1kenobi/cargo-semver-checks/issues/401 for this.

I'm not sure how to pull this off while `cargo-semver-checks` is generating the "current" rustdoc itself, but if generating the current rustdoc separately is on the table then I wrote up an idea in that issue for how that might work. If you're open to exploring it together, I'd be happy to help.